src/app_test.tsx

```diff
@@ -501,3 +501,34 @@ Deno.test("FreshApp - throw when middleware returns no response", async () => {
   expect(res.status).toEqual(500);
   expect(text).toContain("Internal server error");
 });
+
+Deno.test("FreshApp - adding Island should convert to valid export names", () => {
+  const app = new App();
+  const islands = getIslandRegistry(app);
+
+  const component1 = () => <>OK</>;
+  const component2 = () => <>OK</>;
+  const component3 = () => <>OK</>;
+  app.island("/islands/foo.v2.tsx", "default", component1);
+  app.island("/islands/_bar-baz-...-$.tsx", "default", component2);
+  app.island("/islands/1_hello.tsx", "default", component3);
+
+  expect(islands.get(component1)!).toEqual({
+    file: "/islands/foo.v2.tsx",
+    name: "foo_v2",
+    exportName: "default",
+    fn: component1,
+  });
+  expect(islands.get(component2)!).toEqual({
+    file: "/islands/_bar-baz-...-$.tsx",
+    name: "_bar_baz_$",
+    exportName: "default",
+    fn: component2,
+  });
+  expect(islands.get(component3)!).toEqual({
+    file: "/islands/1_hello.tsx",
+    name: "_hello",
+    exportName: "default",
+    fn: component3,
+  });
+});
```

Comment:
I think this is fine.
```suggestion
  const component = () => <>OK</>;
  app.island("/islands/foo.v2.tsx", "default", component);
  app.island("/islands/_bar-baz-...-$.tsx", "default", component);
  app.island("/islands/1_hello.tsx", "default", component);

  expect(islands.get(component1)!).toEqual({
    file: "/islands/foo.v2.tsx",
    name: "foo_v2",
    exportName: "default",
    fn: component,
  });
  expect(islands.get(component2)!).toEqual({
    file: "/islands/_bar-baz-...-$.tsx",
    name: "_bar_baz_$",
    exportName: "default",
    fn: component,
  });
  expect(islands.get(component3)!).toEqual({
    file: "/islands/1_hello.tsx",
    name: "_hello",
    exportName: "default",
    fn: component,
  });
});
```

Reply:
Won't work. Islandregisstry is a map, where the key is the component function. Each one has to be unique, or they overwrite each other. Also, your code has no reference to component1, component2 and component3